rr_evgp/src/UniformCostSearch/UniformCostSearch.cpp

```diff
@@ -0,0 +1,126 @@
+#include <rr_evgp/UniformCostSearch.h>
+#include <algorithm>
+#include <functional>
+#include <opencv2/core/mat.hpp>
+#include <opencv2/core/types.hpp>
+#include <queue>
+#include <vector>
+
+UniformCostSearch::UniformCostSearch(cv::Mat obstacleGrid, cv::Mat distanceGrid, cv::Point startPt, cv::Point goalPt) {
+    obstacleGrid_ = obstacleGrid;
+    distanceGrid_ = distanceGrid;
+    startPoint_ = startPt;
+    goalPoint_ = goalPt;
+}
+
+UniformCostSearch::State UniformCostSearch::getStartState() {
+    UniformCostSearch::State s;
+    s.pt = startPoint_;
+    s.cost = 0.0;
+    return s;
+}
+
+bool UniformCostSearch::isGoalState(UniformCostSearch::State s) {
+    if (s.pt == goalPoint_) {
+        return true;
+    }
+    return false;
+}
+
+float UniformCostSearch::pointToCost(cv::Point pt) {
+    return distanceGrid_.at<float>(pt);  // DistanceTransform outputs CV_32F
+}
+
+bool UniformCostSearch::isValidPoint(cv::Point pt) {
+    cv::Rect rect(cv::Point(), distanceGrid_.size());
+    return rect.contains(pt) && obstacleGrid_.at<uchar>(pt) == UniformCostSearch::FREE_SPACE;
+}
+
+std::vector<UniformCostSearch::State> UniformCostSearch::getSuccessors(UniformCostSearch::State s) {
+    // returns the 8 point grid around a center point
+    std::vector<UniformCostSearch::State> successors;
+    for (int i = -1; i < 2; i++) {
+        for (int j = -1; j < 2; j++) {
+            if (i != 0 || j != 0) {  // not center position
+                cv::Point pt(s.pt.x + i, s.pt.y + j);
+                if (this->isValidPoint(pt)) {
+                    UniformCostSearch::State newState;
+                    newState.pt = pt;
+                    newState.cost = this->pointToCost(pt);
```

Comment:
Not sure why you do `this->function()` instead of just `function()`, I think you do this some other places too